### PR TITLE
Changes for ImageRenderer

### PR DESCRIPTION
**Pull request by sarajaur**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented the Image Item for the model based approach.
- Screenshot 2024-08-25 184309
- https://app.vidcast.io/share/7582907a-ada2-4498-9fa6-10ae24e7f8a3

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

**Review by soumykun — APPROVED** [review]

File: source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        color: imageModel.bgColor != "" ? imageModel.bgColor : "transparent"
+        radius: imageModel.radius
+
+        visible: imageModel.visibleRect
+
+        anchors {
+            left: imageModel.anchorLeft ? parent.left : undefined
+			horizontalCenter: imageModel.anchorCenter ? parent.horizontalCenter : undefined
+			right: imageModel.anchorRight ? parent.right : undefined
+        }
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml:32]: Correct the indentation.

File: source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                x: imageRender.x
+                y: imageRender.y
+            
+                width: imageRender.width
+                height: imageRender.height
+
+                radius: imageRender.radius
+            }
+
+        }
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml:50]: Remove this extra line.

File: source/qml_v2/AdaptiveCardQmlEngine/qml/CollectionItemDelegate.qml [hunk 3]
```diff
@@ -7,24 +7,12 @@ Loader {
...
-    }
-    */
+    Component.onCompleted :{
+		if (model.delegateType == AdaptiveCardEnums.CardElementType.TextBlock)
+			source = "TextBlockRender.qml"
+		else if (model.delegateType == AdaptiveCardEnums.CardElementType.Image)
+			source = "ImageRender.qml"
+        else if (model.delegateType == AdaptiveCardEnums.CardElementType.RichTextBlock)
+            source = "RichTextBlockRender.qml";
+	}
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/CollectionItemDelegate.qml:17]: Correct the indentation.

File: source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    QString mOpenUrl;
+
+    bool mHasAssociatedInputs; 
+    bool mVisibleRect;
+    bool mIsImage;
+    bool mToggleVisibility;
+
+    int mImageHeight;
+    int mImageWidth;
+    int mRadius;
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.h:61]: Check and Cleanup the variables, that are unused.

File: source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp [hunk 5]
```diff
@@ -0,0 +1,135 @@
+#include "ImageModel.h"
+#include "SharedAdaptiveCard.h"
+#include <QDebug.h>
+#include "Utils.h"
+#include "MarkDownParser.h"
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp:6]: Check the header inclusion...is any unused.

File: source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+ImageModel::ImageModel(std::shared_ptr<AdaptiveCards::Image> image, QObject* parent)
+    : QObject(parent),
+    mSourceImage(""), mImageHeight(0), mImageWidth(0), mRadius(0),
+    mIsImage(false), mVisibleRect(false), mBgColor("transparent"),
+    mAnchorCenter(""), mAnchorRight(""), mAnchorLeft("")
+{
+    const auto hostConfig = AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().getHostConfig();
+    const auto rendererConfig = AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().getCardConfig();
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp:16]: If it is not use, please remove it.

File: source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    dir_path.append("\\images\\" + imageName);
+    std::replace(dir_path.begin(), dir_path.end(), '\\', '/');
+    dir_path = "file:/" + dir_path;
+
+    return dir_path;
+}
+
+ImageModel::~ImageModel()
+{
+}
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp:135]: Check for any unused variable and clean it up.

**Review by soumykun — COMMENTED** [review]
Please address comments.

File: source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp (hunk 6 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/models/ImageModel.cpp:15]: Done

File: source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml (hunk 1 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml:32]: Done

File: source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml (hunk 2 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/ImageRender.qml:49]: Done

**Review by soumykun — APPROVED** [review]
LGTM

File: source/qml_v2/AdaptiveCardQmlEngine/qml/CollectionItemDelegate.qml (hunk 3 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/CollectionItemDelegate.qml:17]: Done

**Review by ramkrishnajana — APPROVED** [review]

**Pull request merged**